pkg/resources/validation/namespace/namespace.go

```diff
@@ -0,0 +1,106 @@
+package namespace
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/rancher/webhook/pkg/admission"
+	objectsv1 "github.com/rancher/webhook/pkg/generated/objects/core/v1"
+	"github.com/rancher/webhook/pkg/resources/validation"
+	admissionv1 "k8s.io/api/admission/v1"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	v1 "k8s.io/api/authorization/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	authorizationv1 "k8s.io/client-go/kubernetes/typed/authorization/v1"
+	"k8s.io/utils/trace"
+)
+
+var gvr = schema.GroupVersionResource{
+	Version:  "v1",
+	Resource: "namespaces",
+}
+
+// Validator validates the namespace admission request
+type Validator struct {
+	sar authorizationv1.SubjectAccessReviewInterface
+}
+
+// NewValidator returns a new validator used for validation of namespace requests.
+func NewValidator(sar authorizationv1.SubjectAccessReviewInterface) *Validator {
+	return &Validator{
+		sar: sar,
+	}
+}
+
+// GVR returns the GroupVersionKind for this CRD.
+func (v *Validator) GVR() schema.GroupVersionResource {
+	return gvr
+}
+
+// Operations returns list of operations handled by this validator.
+func (v *Validator) Operations() []admissionregistrationv1.OperationType {
+	return []admissionregistrationv1.OperationType{admissionregistrationv1.Update}
```

Comment:
Should this also cover the Create operation? Currently, anyone with "create" on namespaces can add a PSA label; they just would not be able to update the label after its creation.

Reply:
That might be a reasonable point.  Probably a simple, but important point we missed in design.  I'll circle back on this one soon.